DEVELOPER.md

```diff
@@ -1,3 +1,17 @@
+# Monorepo setup
+
+This repository is a monorepo. All build outputs are created in the `dist` directory.
+
+The `dependencies` in root [`package.json`](package.json) must include all production dependencies
+of the client **and** and the server.
+
+The dependencies of the `server` need to be duplicated in the root [`package.json`](package.json).
+Therefore, if a new prod dependency is added to the server, it needs to be
+added to both [`package.json`](package.json) and [`server/package.json`](server/package.json).
+
+Note that we do not install the dependencies in [`server`](server),
+hence there should not be a `yarn.lock` file.
+
```

Comment:
Does all of the content in the section below still make sense? It was added in the initial commit for the common code: https://github.com/angular/vscode-ng-language-service/commit/fee9166ebeed9291220471983464a935b222d4ae
I think this might change that a little bit? Or do you still need to run `yarn compile`?

Reply:
The content in the section below still stands, because we are still using TypeScript's project references for the build.
Moreover, if you look at `client/tsconfig.json` and `server/tsconfig.json`, the `dist` directory is one of the `rootDirs`, so code in this repo must be compiled to enable project navigation in IDE.